packages/grid_client/src/primitives/network.ts

```diff
@@ -119,11 +122,63 @@ class Network {
     return this.wireguardConfig;
   }
 
-  async addNode(node_id: number, metadata = "", description = "", subnet = ""): Promise<Workload | undefined> {
-    if (this.nodeExists(node_id)) {
+  // Check if network will be updated with mycelium or not
+  async checkMycelium(nodeId: number, mycelium: boolean, myceliumSeeds: MyceliumNetworkModel[] = []) {
+    if (!mycelium) return;
+    // check if network has mycelium or not
+    const network = this.networks.find(network => {
+      return network["node_id"] === nodeId;
+    });
+    const myceliumNetworkSeed = myceliumSeeds.find(item => item.nodeId == nodeId);
+    if (network && network.mycelium && network.mycelium?.hex_key) {
+      if (myceliumSeeds && myceliumNetworkSeed?.seed !== network.mycelium.hex_key) {
```

Comment:
mycelium seeds now is a list so just checking the variable name is not enough, we need to check its length too.